qiskit/providers/aer/pulse_new/models/signals.py

```diff
@@ -211,3 +212,96 @@ def signal_add(sig1, sig2):
                           self.carrier_freq)
         else:
             return Signal(lambda t: sig1.value(t) + sig2.value(t))
+
+
+class VectorSignal:
+    """The vector version of the Signal class - the envelope is an array-valued
+    function, and carrier_freqs is an array of carrier frequencies.
+
+    In addition, a drift_array is set to correspond to the value of the
+    VectorSignal when all "time-dependent terms" are off. E.g. if it is
+    composed of a list of Signal objects, this corresponds to the output when
+    all non-Constant signal objects are zero.
+    """
+
+    def __init__(self,
+                 envelope,
+                 carrier_freqs,
+                 drift_array=None):
+        """Initialize.
+
+        Args:
+            envelope (Callable): function of a single float returning an array.
+            carrier_freqs (array): list of carrier frequences
+            drift_array (array): the drift array
```

Comment:
Needs more information.

Reply:
Tried adding some here - there is also more info in the class doc string